assets/js/main.js

```diff
@@ -0,0 +1,69 @@
+// Add Repository modal handling
+window.CAN_USE_HEADERS = location.hostname.includes('baz.ninja');
+
+const modal = document.getElementById('add-repo-modal');
+const openBtn = document.getElementById('add-repo-card');
+const closeBtn = modal.querySelector('.modal__close');
+const backdrop = modal.querySelector('.modal__backdrop');
+const form = document.getElementById('add-repo-form');
+const feedbackEl = document.getElementById('repo-add-feedback');
+
+// Open modal
+openBtn && openBtn.addEventListener('click', () => {
+  modal.classList.add('modal--open');
+  feedbackEl.textContent = '';
+  form.reset();
+});
+
+// Close modal
+[closeBtn, backdrop].forEach(el =>
+  el.addEventListener('click', () => modal.classList.remove('modal--open'))
+);
+
+// Submit handler
+form && form.addEventListener('submit', async e => {
+  e.preventDefault();
+  feedbackEl.textContent = '';
+  const url = document.getElementById('repo-url-input').value.trim();
+  const m = url.match(/github\.com\/([^/]+\/[^/]+)(?:\/|$)/);
+  if (!m) {
+    feedbackEl.textContent = 'Repo not valid';
```

Comment:
<p>Let's add an example of what a url should look like. Also, org/repo should be valid</p>